include/clasp/core/array.h

```diff
@@ -224,7 +226,7 @@ class Array_O : public General_O {
   /*! length() doesn't dispatch - it reaches into the subclass
       through the _Length[0] array to read the first size_t element
       which is the Length/FillPointer for vectors and a Dummy value for arrays */
-  size_t length() const { return this->_Length[0]; };
+  virtual size_t length() const { return this->_Length[0]; };
```

Comment:
I don't understand this change. Why does this have to work differently for short/long arrays? Is there some way for _Length[0] and the gcarray length to get out of sync? If I remember correctly, the idea here is that for a 1d simple array _Length[0] reaches into the underlying gcarray anyway.

If this change is necessary, the above comment saying "length() doesn't dispatch" needs to be changed, since if length is virtual the comment is false.

Reply:
I didn't want make the change, but I'm not sure what to do. I think that because x86 stores binary80 with 6 bytes of padding that affects the alignment of the structure so the non-dispatching hack doesn't work anymore. I'm not sure why we have such a hack anyways?

Reply:
well the hack is basically so that getting the length of an array is a simple memory access regardless of type. If the length is no longer in that position we might also have to rejigger the compiler; at least for other simple arrays it assumes it can just grab the length without calling a function.

Reply:
Fixed with packed attribute.